src/core/MOM.F90

```diff
@@ -2571,8 +2581,10 @@ subroutine initialize_MOM(Time, Time_init, param_file, dirs, CS, restart_CSp, &
     endif
   endif
 
-  if (use_ice_shelf .and. CS%debug) &
+  if (use_ice_shelf .and. CS%debug) then
     call hchksum(CS%frac_shelf_h, "MOM:frac_shelf_h", G%HI, haloshift=0)
+    call hchksum(CS%mass_shelf, "MOM:mass_shelf", G%HI, haloshift=0)
```

Comment:
This checksum call form mass_shelf needs to have a scaling factor of `scale=US%RZ_to_kg_m2`